dre_repop_adept_sessions.py

```diff
@@ -0,0 +1,130 @@
+import requests
+
+from pymongo import MongoClient
+from decouple import config
+
+MONGO_URL = config('MONGO_URL')
+
+# PROD TA1 outside AWS
+# ADEPT_URL = "https://darpaitm.caci.com/adept/"
+# ST_URL = "https://darpaitm.caci.com/soartech/" 
+
+#DEV
+ADEPT_URL="http://localhost:8081/"
+# ST_URL="http://localhost:8084/"
+
+
+# PROD TA1 inside AWS
+# ADEPT_URL="http://10.216.38.101:8080/"
+# ST_URL="http://10.216.38.125:8084"
+
+
+def update_adept_text_adm_sessions(mongoDB):
+    text_scenario_collection = mongoDB['userScenarioResults']
+    text_scenario_to_update = text_scenario_collection.find({"evalNumber": 4})
+    adm_collection = mongoDB["test"]
+    adms_to_update = adm_collection.find({"evalNumber": 4})
+
+    # Add text sessions to Adept Server
+    sessions_by_pid = {}
+    for entry in text_scenario_to_update:
+        scenario_id = entry.get('scenario_id')
+        session_id = entry.get('combinedSessionId', entry.get('serverSessionId'))
+        data_id = entry.get('_id')
+        pid = entry.get('participantID')
+        kdmas = requests.get(f'{ADEPT_URL}api/v1/computed_kdma_profile?session_id={session_id}').json()
+        
+        # ADEPT's server lost our first week's worth of data, so we have to re-send probes
+        new_id = None
+        if 'DryRun' in scenario_id and 'value' not in kdmas:
+            if pid in sessions_by_pid:
+                new_id = sessions_by_pid[pid]['sid']
+                sessions_by_pid[pid]['_ids'].append(data_id)
+            else:
+                adept_sid = requests.post(f'{ADEPT_URL}/api/v1/new_session').text.replace('"', '').strip()
+                sessions_by_pid[pid] = {'sid': adept_sid, '_ids': [data_id]}
+                new_id = adept_sid
+            # collect probes
+            probes = []
+            for k in entry:
+                if isinstance(entry[k], dict) and 'questions' in entry[k]:
+                    if 'probe ' + k in entry[k]['questions'] and 'response' in entry[k]['questions']['probe ' + k] and 'question_mapping' in entry[k]['questions']['probe ' + k]:
+                        response = entry[k]['questions']['probe ' + k]['response']
+                        mapping = entry[k]['questions']['probe ' + k]['question_mapping']
```

Comment:
THIS NEEDS TO CHANGE!! In mongo, responses are NOT stored exactly as the mapping keys, which means we are currently NOT sending all adept probes when we run this, meaning everything is wrong. 
                                    
                                    response = entry[k]['questions']['probe ' + k]['response'].replace('.', '')
                                    mapping = entry[k]['questions']['probe ' + k]['question_mapping']
                                    if response in mapping:
                                        probes.append({'probe': {'choice': mapping[response]['choice'], 'probe_id': mapping[response]['probe_id']}})
                                    else:
                                        print('could not find response in mapping!', response, list(mapping.keys()))

Reply:
Added fix: tested pre and post,

added probe outputs:
`python dre_repop_adept_sessions.py > run.txt`

Pre:
There were 11 with txt calls with empty probes, found a couple with incomplete probes (probably more, jsut spot checking)

Post:
No empty probes